### PR TITLE
deleted mdy function from reimb builder

### LINKED ISSUE #514: MNT: move mdy function from reimb builder to dates and add tests

### PR DESCRIPTION
**Pull request by dy2403**
Closes #514 
Hi prof. @sbillinge,
I moved the mdy function from reimb builder to dates and added tests.
Can you please review it?

### CODE REVIEW

File: regolith/dates.py [hunk 1]
```diff
@@ -420,3 +420,28 @@ def is_between(thing, start=None, end=None):
...
+def mdy(month, day, year, **kwargs):
+    """
+    Given a month, day, and year, returns a string with the date
+    in the following format: MM/DD/YY
+
+    Parameters
+    ----------
+    month: integer or string
+         the month.  if a string should be resolvable using regolith month_to_int
+
```
Comment (sbillinge) [inline regolith/dates.py:433]: remove empty lines.  If you want to see the format, please google for "numpy docstrings"

**Comment by sbillinge** [comment]
Actuallly, come to think of it, I think we don't need this at all because if we are passing dates around we can use ``strftime`` method, so maybe just remove this function from everywhere it is used and replace with strftime

File: regolith/builders/reimbursementbuilder.py [hunk 2]
```diff
@@ -118,9 +109,9 @@ def excel(self):
                             se_column = 14
                         dates.append(expdates.get("date"))
                         item_ws.cell(row=r, column=2, value=i)
-                        item_ws.cell(row=r, column=3, value=mdy(expdates.get("date").month,
-                                              expdates.get("date").day,
-                                              expdates.get("date").year))
+                        item_ws.cell(row=r, column=3, value=dt.datetime(int(expdates.get("date").year),
```
Comment (sbillinge) [inline regolith/builders/reimbursementbuilder.py:112]: doesn't 
```
dt.datetime(expdates.get("date)).strftime('%x')
```
work?

Comment (sbillinge) [reply regolith/builders/reimbursementbuilder.py:112]: also I see the same mdy function being used in coa builder.  Please could you do the same cleaning there as well?  Thx.

**Comment by dy2403** [comment]
@sbillinge I am trying to understand what gets as input to this builder and how tests work to builders. Thus, I am not fully confident of what I am doing here.

File: regolith/builders/reimbursementbuilder.py (hunk 2 above)
Comment (dy2403) [reply regolith/builders/reimbursementbuilder.py]: I think the mdy function was defined there, but wasn't used. Thus, I just deleted it.

Comment (sbillinge) [reply regolith/builders/reimbursementbuilder.py]: cool.  Remove WIP and request merge if you are ready.  Make sure there is a news.

**Comment by sbillinge** [comment]
> 
> 
> @sbillinge I am trying to understand what gets as input to this builder and how tests work to builders. Thus, I am not fully confident of what I am doing here.
> Here, I am treating dates as date time objects, which I guess they are because min(dates) and max(dates) are used, is it correct?

yes, correct.

File: regolith/dates.py (hunk 1 above)
Comment (dy2403) [reply regolith/dates.py]: this was deleted!

**Comment by sbillinge** [comment]
Thanks @dy2403

**Pull request merged**